pypesto/sample/parallel_tempering.py

```diff
@@ -197,12 +198,27 @@ def compute_log_evidence(
                 f"Carefully check the results. Consider using beta_init='{BETA_DECAY}' for better results."
             )
 
-        burn_in = result.sample_result.burn_in
-        trace_loglike = (
-            result.sample_result.trace_neglogprior[::-1, burn_in:]
-            - result.sample_result.trace_neglogpost[::-1, burn_in:]
-        )
-        mean_loglike_per_beta = np.mean(trace_loglike, axis=1)
+        # compute burn in for all chains and then estimate mean of log likelihood for each beta
+        mean_loglike_per_beta = []
+        temps = []
+        for i_chain in reversed(range(len(self.betas))):
+            burn_in_i = geweke_test(result, chain_number=i_chain)
+
+            if (
+                burn_in_i < result.sample_result.trace_x.shape[1]
+                or i_chain == len(self.betas) - 1
```

Comment:
Can this be changed to `self.betas[i_chain] == 0`?

Should the Geweke be skipped for this chain, because you already know it's converged?
I would either skip the Geweke test for this chain, or remove this condition from the `if`.

Reply:
I can check that. For `beta` equal to 0, we only have samples from the prior. So it has always converged, there is no burn in.

Reply:
Then also skip the geweke test if `beta == 0`